build_scripts/versions.yaml

```
@@ -23,6 +23,16 @@ default-pg-max: 15
 default-cargo-pgx: 0.6.1
 default-arch: both
 
+## this contains the specific pg versions that will be installed.
+## from https://www.postgresql.org/about/news/postgresql-172-166-1510-1415-1318-and-1222-released-2965/
+postgres_versions:
+  - 17.2
+  - 16.6
+  - 15.10
+  - 14.15
+  - 13.18
+  - 12.22
```

I think this is a little nicer to work with:
```suggestion
  17: 17.2
  16: 16.6
  15: 15.10
  14: 14.15
  13: 13.18
  12: 12.22
```

Then your query for versions could be just:
```
PINNED=$(yq ".postgres_versions.${MAJOR}" /build/scripts/versions.yaml
```

(which is simple enough to possibly not bother with an extra .sh script.